### ISSUE TITLE
nvda is not announcing when a cell in excel contains formula 

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#13457]
### Steps to reproduce:
in nvda the option to use uia for excel is disabled.
open a spreadsheet in excel and enter a formula
now place the arrows focus on the cell with the formula. 
### Actual behavior:
nvda does not report that the cell has a formula 
### Expected behavior:
nvda should announce if the cell has a formula.
I remember this working a while ago, I don't know who caused the problem, if an nvda or office update.
If you check the option to use uia for excel in nvda's advanced options everything works fine.
I still don't allow the use of uia in excel because it still doesn't allow the use of the navigation mode and the list of elements. 
#### NVDA installed/portable/running from source:
instaled
#### NVDA version:
nvda.exe, NVDA alpha-24917,12319e7b

#### Windows version:
11 22000.556
#### Name and version of other software in use when reproducing the issue:
EXCEL.EXE, Microsoft Office 16.0.14931.20132

#### Does the issue still occur after restarting your computer?
yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no
#### If NVDA add-ons are disabled, is your problem still occurring?
yes
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
yes

**Comment by CyrilleB79** [comment]
Yes, it is confirmed also on my side with Excel 2016 version '16.0.14931.20120':
* NVDA run from source (last master = 6e39f7f7b7b7858e10d06b22c1c82dff7a3cb425): 'has formula' is not reported.
* NVDA 2021.3.3 installed: 'has formula' is reported.

Thus it seems to be a regression during 2022.1 alpha stage.

Cc @feerrenrut, @michaelDCurran

**Comment by michaelDCurran** [comment]
Caused by merge of pr #13414 
In-proc code for Excel contains some state constants which should mirror NvDA's controlTypes state constants. 
However, with the merge of that pr, the state constants no longer match.

**Comment by abdel792** [comment]
Hi @CyrilleB79 and all,

In fact, the Thunderbird+ add-on has been around for a long time, and several people contributed to it.

I'd proposed my help to Pierre-Louis only for some additional features he needed.

I don't think Pierre-Louis is on GitHub.

Regarding the use of numeric values for roles, they were introduced by other developers who preceded my work on this add-on.

If you open Windows Notepad and then NVDA's Python console, a statement like the following should show True.

> focus.role == 8

I think this is the type of comparison you mean?

Thanks.

**Comment by CyrilleB79** [comment]
Please change milestone to 2022.1 instead of 2022.2.
